runtime/tests/interpreter/interpreter_test.go

```diff
@@ -10741,6 +10741,209 @@ func TestInterpretArrayReverse(t *testing.T) {
 	}
 }
 
+func TestInterpretArrayFilter(t *testing.T) {
+	t.Parallel()
+
+	inter := parseCheckAndInterpret(t, `
+		let xs = [1, 2, 3, 100, 201]
+		let xs_fixed: [Int; 5] = [1, 2, 3, 100, 201]
+		let emptyVals: [Int] = []
+		let emptyVals_fixed: [Int; 0] = []
+
+		let onlyEven =
+			fun (_ x: Int): Bool {
+				return x % 2 == 0
+			}
+
+		fun filterxs(): [Int] {
+			return xs.filter(onlyEven)
+		}
+		fun originalxs(): [Int] {
+			return xs
+		}
+
+		fun filterxs_fixed(): [Int] {
+			return xs_fixed.filter(onlyEven)
+		}
+		fun originalxs_fixed(): [Int; 5] {
+			return xs_fixed
+		}
+
+		fun filterempty(): [Int] {
+			return emptyVals.filter(onlyEven)
+		}
+		fun originalempty(): [Int] {
+			return emptyVals
+		}
+
+		fun filterempty_fixed(): [Int] {
+			return emptyVals_fixed.filter(onlyEven)
+		}
+		fun originalempty_fixed(): [Int; 0] {
+			return emptyVals_fixed
+		}
+
+		pub struct TestStruct {
+			pub var test: Int
+
+			init(_ t: Int) {
+				self.test = t
+			}
+		}
+
+		let onlyOddStruct =
+			fun (_ x: TestStruct): Bool {
+				return x.test % 2 == 1
+			}
+
+		let sa = [TestStruct(1), TestStruct(2), TestStruct(3)]
+		let sa_fixed: [TestStruct; 3] = [TestStruct(1), TestStruct(2), TestStruct(3)]
+		
+		fun filtersa(): [Int] {
+			let sa_filtered = sa.filter(onlyOddStruct)
+			let res: [Int] = [];
+			for s in sa_filtered {
+				res.append(s.test)
+			}
+			return res
+		}
+		fun originalsa(): [Int] {
+			let res: [Int] = [];
+			for s in sa {
+				res.append(s.test)
+			}
+			return res
+		}
+
+		fun filtersa_fixed(): [Int] {
+			let sa_rev = sa_fixed.filter(onlyOddStruct)
+			let res: [Int] = [];
+			for s in sa_rev {
+				res.append(s.test)
+			}
+			return res
+		}
+		fun originalsa_fixed(): [Int] {
+			let res: [Int] = [];
+			for s in sa_fixed {
+				res.append(s.test)
+			}
+			return res
+		}
+	`)
```

Comment:
This test case is testing many different cases. Can you maybe split this into multiple tests, with each one testing only a particular scenario? It does make sense to combine testing/asserting the 'filtered' array and the 'original' array into one test, but testing fixed vs variable-sized arrays, empty arrays, and struct arrays, can be different test cases.

Reply:
Done, refactored tests into multiple smaller ones.